AK/Random.cpp

```diff
@@ -1,98 +1,126 @@
 /*
  * Copyright (c) 2021, the SerenityOS developers.
+ * Copyright (c) 2024, the Ladybird developers.
+ * Copyright (c) 2025, Colleirose <criticskate@pm.me>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/Assertions.h>
 #include <AK/Platform.h>
 #include <AK/Random.h>
 #include <AK/UFixedBigInt.h>
 #include <AK/UFixedBigIntDivision.h>
 
+#if defined(AK_OS_LINUX)
+#    include <sys/random.h>
+#endif
+
 #if defined(AK_OS_WINDOWS)
 #    include <AK/NumericLimits.h>
 #    include <AK/Windows.h>
 #    include <bcrypt.h>
 #    include <ntstatus.h>
+#    include <winternl.h>
 #endif
 
-namespace AK {
-
-// NOTE: This function is supposed to always give a random number. If possible it is of good quality, but it can fall
-//       back to rand() if it fails on some systems. For high speed you should probably use a different generator.
-//       See MathObject::random() from LibJS. Where cryptographic security is needed use LibCrypto/SecureRandom.h.
-void fill_with_random([[maybe_unused]] Bytes bytes)
+static inline ErrorOr<void> csprng(void* const buf, size_t size)
```

Comment:
Why is this `void* const buf`? You're immediately `const_cast`ing it in the Windows case.